File: schedule_front/lib/features/schedule/domain/models/reminder.dart
class Reminder {
  final String? reminderId;
  final String reminder_title;
  final String recurrenceDays;
  final int? reminderMinutesBefore;
  final bool? isActive;
  final String? date;
  final DateTime? checkedDate;
  final DateTime? recurrenceStartDate;
  final DateTime? recurrenceEndDate;
  final String? excludedDates;

  Reminder({
    this.reminderId,
    required this.reminder_title,
    required this.recurrenceDays,
    this.reminderMinutesBefore,
    this.isActive = true,
    this.date,
    this.checkedDate,
    this.recurrenceStartDate,
    this.recurrenceEndDate,
    this.excludedDates,
  });

  Reminder copyWith({
    String? reminderId,
    String? reminder_title,
    String? recurrenceDays,
    int? reminderMinutesBefore,
    bool? isActive,
    String? date,
    DateTime? checkedDate,
    DateTime? recurrenceStartDate,
    DateTime? recurrenceEndDate,
    String? excludedDates,
  }) {
    return Reminder(
      reminderId: reminderId ?? this.reminderId,
      reminder_title: reminder_title ?? this.reminder_title,
      recurrenceDays: recurrenceDays ?? this.recurrenceDays,
      reminderMinutesBefore: reminderMinutesBefore ?? this.reminderMinutesBefore,
      isActive: isActive ?? this.isActive,
      date: date ?? this.date,
      checkedDate: checkedDate ?? this.checkedDate,
      recurrenceStartDate: recurrenceStartDate ?? this.recurrenceStartDate,
      recurrenceEndDate: recurrenceEndDate ?? this.recurrenceEndDate,
      excludedDates: excludedDates ?? this.excludedDates,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (reminderId != null) 'id': reminderId,
      'reminderTitle': reminder_title,
      'recurrenceDays': recurrenceDays,
      'reminderMinutesBefore': reminderMinutesBefore,
      'isActive': isActive,
      'date': date,
      if (checkedDate != null) 'checkedDate': checkedDate!.toIso8601String(),
      if (recurrenceStartDate != null) 'recurrenceStartDate': recurrenceStartDate!.toIso8601String(),
      if (recurrenceEndDate != null) 'recurrenceEndDate': recurrenceEndDate!.toIso8601String(),
      'excludedDates': excludedDates,
    };
  }

  factory Reminder.fromJson(Map<String, dynamic> json) {
    return Reminder(
      reminderId: json['id']?.toString() ?? json['reminderId'] as String?,
      reminder_title: json['reminderTitle'] as String,
      recurrenceDays: json['recurrenceDays'] as String,
      reminderMinutesBefore: json['reminderMinutesBefore'] as int?,
      isActive: json['isActive'] as bool?,
      date: json['date'] as String?,
      checkedDate: json['checkedDate'] != null 
          ? DateTime.parse(json['checkedDate'] as String) 
          : null,
      recurrenceStartDate: json['recurrenceStartDate'] != null 
          ? DateTime.parse(json['recurrenceStartDate'] as String) 
          : null,
      recurrenceEndDate: json['recurrenceEndDate'] != null 
          ? DateTime.parse(json['recurrenceEndDate'] as String) 
          : null,
      excludedDates: json['excludedDates'] as String?,
    );
  }
} 

File: schedule_back/backend_schedule_04_gradle_plus/src/main/java/org/example/domain/reminder/Reminder.java
package org.example.domain.reminder;

import jakarta.persistence.*;
import lombok.*;
import org.example.domain.user.User;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Reminder {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "reminder_id")
    private Long id;

    @Column(name = "reminder_title", nullable = false)
    private String reminderTitle;

    @Column(name = "recurrence_days", nullable = false)
    private String recurrenceDays;

    @Column(name = "reminder_minutes_before", nullable = true)
    private Integer reminderMinutesBefore;

    @Column(name = "reminder_time", nullable = false)
    private LocalDateTime reminderTime;  // 실제 알림 시간

    @Column(name = "is_active", nullable = false)
    @Builder.Default
    private Boolean isActive = true;     // 활성화 여부 (기본 true)

    @Column(name = "checked_date", nullable = true)
    private LocalDateTime checkedDate;   // 체크된 날짜 (null이면 체크되지 않음)

    @Column(name = "recurrence_start_date", nullable = true)
    private LocalDateTime recurrenceStartDate;

    @Column(name = "recurrence_end_date", nullable = true)
    private LocalDateTime recurrenceEndDate;
    
    @Column(name = "excluded_dates", nullable = true, columnDefinition = "TEXT")
    private String excludedDates; // 쉼표로 구분된 ISO 날짜 문자열 "2023-10-01,2023-10-08"

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", referencedColumnName = "firebase_uid")
    private User user;
}
